test/integration/canonicalize/blocks.stan

```diff
@@ -0,0 +1,20 @@
+data {
+  int N;
+  vector[N] y;
+}
+parameters {
+   real mu;
+   real<lower=0> sigma;
+}
+model {
+    int j = 0;
+
+    if (N>5) y ~ normal(mu, sigma);
```

Comment:
Can you add some more complicated examples here? What happens if you have nested ifs/fors, or a comment between the if and the statement, or an if/else that has brackets for the if but not for the else?


Feel free to make the test file pretty ugly to really stress this feature out